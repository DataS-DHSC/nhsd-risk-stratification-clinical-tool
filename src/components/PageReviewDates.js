import React from 'react';
import PropTypes from 'prop-types';
import { ReviewDate } from 'nhsuk-react-components';
import { PAGE_REVIEW_DATES } from '../constants/reviewDates';

const PageReviewDates = ({ page }) => {
  const dates = PAGE_REVIEW_DATES[`${page}`] || {
    lastReviewDate: '',
    nextReviewDate: '',
  };

  return (
    <>
      {(dates.lastReviewDate || dates.nextReviewDate) && (
        <ReviewDate
          id="review-date"
          lastReviewed={dates.lastReviewDate}
          nextReview={dates.nextReviewDate}
        />
      )}
    </>
  );
};

PageReviewDates.propTypes = {
  page: PropTypes.string.isRequired,
};

export default PageReviewDates;
