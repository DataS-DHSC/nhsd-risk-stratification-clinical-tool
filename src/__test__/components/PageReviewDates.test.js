import React from 'react';
import { render } from '@testing-library/react';
import PageReviewDates from '../../components/PageReviewDates';

const setup = (page) => render(<PageReviewDates page={page} />);

describe('PageReviewDates', () => {
  it('should match snapshot for static pages', () => {
    const container = setup('static');
    expect(container.asFragment()).toMatchSnapshot();
  });

  it('should match snapshot for qcovid pages', () => {
    const container = setup('qcovid');
    expect(container.asFragment()).toMatchSnapshot();
  });
});
