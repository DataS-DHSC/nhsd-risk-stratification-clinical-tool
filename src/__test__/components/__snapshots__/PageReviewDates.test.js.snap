// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageReviewDates should match snapshot for qcovid pages 1`] = `
<DocumentFragment>
  <div
    class="nhsuk-review-date"
    id="review-date"
  >
    <p
      class="nhsuk-body-s"
    >
      Page last reviewed: 1 December 2020
      <br />
      Next review due: 23 December 2023
    </p>
  </div>
</DocumentFragment>
`;

exports[`PageReviewDates should match snapshot for static pages 1`] = `
<DocumentFragment>
  <div
    class="nhsuk-review-date"
    id="review-date"
  >
    <p
      class="nhsuk-body-s"
    >
      Page last reviewed: 1 December 2020
      <br />
      Next review due: 23 December 2023
    </p>
  </div>
</DocumentFragment>
`;
